lighthouse-core/lib/page-functions.js

```diff
@@ -409,12 +409,14 @@ function getNodeLabel(node) {
 
 /**
  * @param {HTMLElement} element
- * @return {LH.Artifacts.Rect}
+ * @return {LH.Artifacts.Rect|undefined}
  */
 /* istanbul ignore next */
 function getBoundingClientRect(element) {
   // The protocol does not serialize getters, so extract the values explicitly.
   const rect = element.getBoundingClientRect();
+  if (rect.width === 0 || rect.height === 0) return;
```

Comment:
other places use this function and might expect a 0x0 rect returned, though...

Reply:
darn @ts-expect-error and stringification! I'll look closer.

Reply:
nope, this is only used in two places: page-function getNodeDetails, and the FPS gatherer.

Reply:
Can the silliness be avoided through other means like ignoring `head` contents?

I agree that doing this for *all* elements that happen to not have size might be a mistake. The location can often still provide value in hidden element cases.

Reply:
yeah, I guess I meant a function called `getBoundingClientRect` seems like the wrong place to determine if it's worth keeping the rect from the element or not. `getBoundingClientRectIfItHasSize`, maybe :)

Agreed it is a little weird that _all_ the uses of NodeDetails keep their `boundingRect`, though. e.g. `LinkElements` didn't keep them before #11405. Maybe we need an easy way for callers of `getNodeDetails` to indicate that they know that 0x0 bounding rects aren't important and can be discarded?

Reply:
points well taken! I've reversed the approach here.